clients/src/main/java/org/apache/kafka/common/security/authenticator/SaslClientAuthenticator.java

```diff
@@ -524,10 +524,14 @@ private AbstractResponse receiveKafkaResponse() throws IOException {
                 currentRequestHeader = null;
                 return response;
             }
-        } catch (SchemaException | IllegalArgumentException e) {
+        } catch (SchemaException | IllegalStateException | IllegalArgumentException e) {
```

Comment:
the side effect is that the mismatched error about correlation id is added back to ```Selector``` so I'm digging in it for better approach.